### DISCUSSION TITLE
iModel Query suspension error

### DISCUSSION THREAD

**Discussion by pateljay1397** [iTwin/itwinjs-core#4162] [Q&A] [answered]
I am running following query to fetch the data from the iModel("**Stadium**"):
**Query**: `select ECInstanceID,Origin,Yaw,Pitch,Roll,BBoxLow,BBoxHigh from biscore.physicalelement where category.id in (select ecinstanceid from biscore.category where codevalue like '%SS_Pitch_Grass%')`;

While running this query from the iModelConsole I am getting the expected results but when I run the same query from the code it is giving me query suspended error. Please let me know if you need any further information on this. Attached the code snippet for your reference:

```
    if(iModelConnection && iModelConnection.isOpen){
      const query = `select ECInstanceID,Origin,Yaw,Pitch,Roll,BBoxLow,BBoxHigh from biscore.physicalelement where category.id in (select ecinstanceid from biscore.category where codevalue like '%SS_Pitch_Grass%')`;
      const results = iModelConnection.query(query, undefined, { rowFormat: QueryRowFormat.UseJsPropertyNames });
      console.log("Results",results);
    }
```

**Comment by aruniverse** [comment, marked as answer]
IModelConnection.query returns an AsyncIterator, see the [docs](https://www.itwinjs.org/reference/core-frontend/imodelconnection/imodelconnection/query/)

You need to do something like:

```ts
  const iter = iModelConn.query(query, undefined, {
    rowFormat: QueryRowFormat.UseJsPropertyNames,
  });

  const rows = [];
  for await (const row of iter) {
    rows.push(row);
  }

  console.log(rows);
```

**Reply by pateljay1397** [reply]
Thank you @aruniverse  for your prompt response it has resolved the issue and I appreciate you for the explanation.
